lib/DBD/mysql.pm

```diff
@@ -1186,6 +1186,18 @@ flags:
 These are used to specify the respective parameters of a call
 to mysql_ssl_set, if mysql_ssl is turned on.
 
+=item mysql_ssl_verify_server_cert
+
+This turns on the MYSQL_OPT_SSL_VERIFY_SERVER_CERT connection option.
+
+Note that this is deprecated as of MySQL 5.7.11 and is removed in MySQL 8.0
+
+=item mysql_ssl_mode
+
+Can be set to 'disabled', 'preferred', 'required', 'verify_ca' or 'verify_identity'.
+Default is 'preferred'.
+
+Available from MySQL 5.7.11
```

Comment:
Seems that it is not possible to verify if connection is encrypted before sending mysql password :-(

According to https://dev.mysql.com/doc/refman/5.6/en/mysql-ssl-set.html it is possible to verify if connection is encrypted or not via `mysql_get_ssl_cipher(sock)` -- but only after `mysql_real_connect(sock, host, user, passwd, ...)` was called. Which could send password over non encrypted channel.

It is better than nothing. But should be written DBD::mysql documentation with big warning!